library/net.c

```diff
@@ -203,7 +207,11 @@ int mbedtls_net_bind( mbedtls_net_context *ctx, const char *bind_ip, const char
 
         n = 1;
         if( setsockopt( ctx->fd, SOL_SOCKET, SO_REUSEADDR,
+#if VXWORKS
+                        (char *) &n, sizeof( n ) ) != 0 )
+#else
```

Comment:
I'm saying that the `(char*)` cast should be accepted by nonmal compilers (I think?), so we shouldn't need an ifdef, the `#if VXWORKS` lines here could well work on all platforms (it's worth a try).